apps/web/src/services/analytics/mixpanel-tracking.ts

```diff
@@ -0,0 +1,305 @@
+import mixpanel from 'mixpanel-browser'
+import { IS_PRODUCTION } from '@/config/constants'
+import { useHasFeature } from '@/hooks/useChains'
+import { FEATURES } from '@safe-global/utils/utils/chains'
+import type { SafeUserAttributes, MixPanelUserProfileUpdate, SafeEventProperties } from './types'
+import { prepareMixPanelUserAttributes } from './user-attributes'
+
+/**
+ * Checks if MixPanel is available and properly initialized
+ * @returns boolean indicating if MixPanel is ready
+ */
+const isMixPanelReady = (): boolean => {
+  if (typeof window === 'undefined') return false
+  if (!mixpanel) return false
+
+  try {
+    // Check if MixPanel is initialized by checking for required methods
+    return (
+      typeof mixpanel.track === 'function' &&
+      typeof mixpanel.people === 'object' &&
+      typeof mixpanel.identify === 'function'
```

Comment:
Why do we have to check the types of these attributes in order to decide if mixpanel is ready? Isn't `if (!mixpanel) return false` above sufficient?